### PR TITLE
Remove mac specific skips

### PR DESCRIPTION
**Pull request by stefannikolei**
### Prerequisites

- [ ] I have written a descriptive pull-request title
- [ ] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [ ] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [ ] I have provided test coverage for my change (where applicable)

### CODE REVIEW

**Comment by stefannikolei** [comment]
Strange. 

Locally all tests work on my Mac..

**Comment by JimBobSquarePants** [comment]
The VMs are super low spec here so sometimes the operations we expect (GC cleanup) simply do not occur in the timescale. 

It's easier just to test on a Mac locally and move on.

**Comment by JimBobSquarePants** [comment]
Looks like we can allow the Tiff decoder tests now but I would keep the MemoryAllocator ones diabled for now.

File: tests/ImageSharp.Tests/Memory/Allocators/UniformUnmanagedMemoryPoolTests.Trim.cs [hunk 1]
```diff
@@ -58,6 +58,7 @@ public class NonParallel
             // TODO: Investigate failures on macOS. All handles are released after GC.
             // (It seems to happen more consistently on .NET 6.)
             [ConditionalFact(nameof(IsNotMacOS))]
+            [Fact]
```
Comment (stefannikolei) [inline tests/ImageSharp.Tests/Memory/Allocators/UniformUnmanagedMemoryPoolTests.Trim.cs:61]:
```suggestion
```

File: tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs [hunk 2]
```diff
@@ -202,13 +202,13 @@ public void TiffDecoder_CanDecode_12Bit_WithUnassociatedAlpha<TPixel>(TestImageP
...
-            image.DebugSave(provider);
-            return;
-        }
+        // if (TestEnvironment.IsMacOS)
+        // {
+        //     // Only debug save on OSX: For some reason the reference image has a difference of 50%. The imagesharp output file looks ok though.
+        //     using Image<TPixel> image = provider.GetImage(TiffDecoder.Instance);
+        //     image.DebugSave(provider);
+        //     return;
+        // }
```
Comment (stefannikolei) [inline tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs:211]:
```suggestion
```

File: tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs [hunk 3]
```diff
@@ -258,13 +251,13 @@ public void TiffDecoder_CanDecode_20Bit_WithAssociatedAlpha<TPixel>(TestImagePro
...
```
Comment (stefannikolei) [inline tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs:260]:
```suggestion
```

File: tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs [hunk 4]
```diff
@@ -284,13 +270,13 @@ public void TiffDecoder_CanDecode_24Bit_WithUnassociatedAlpha<TPixel>(TestImageP
...
```
Comment (stefannikolei) [inline tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs:279]:
```suggestion
```

File: tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs [hunk 5]
```diff
@@ -385,13 +364,13 @@ public void TiffDecoder_CanDecode_32Bit_WithUnassociatedAlpha<TPixel>(TestImageP
...
```
Comment (stefannikolei) [inline tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs:373]:
```suggestion
```

File: tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs [hunk 6]
```diff
@@ -425,13 +404,13 @@ public void TiffDecoder_CanDecode_40Bit_WithUnassociatedAlpha<TPixel>(TestImageP
...
```
Comment (stefannikolei) [inline tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs:413]:
```suggestion
```

File: tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs [hunk 7]
```diff
@@ -463,13 +442,13 @@ public void TiffDecoder_CanDecode_48Bit_WithUnassociatedAlpha<TPixel>(TestImageP
...
```
Comment (stefannikolei) [inline tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs:451]:
```suggestion
```

File: tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs [hunk 8]
```diff
@@ -492,13 +471,13 @@ public void TiffDecoder_CanDecode_56Bit_WithUnassociatedAlpha<TPixel>(TestImageP
...
```
Comment (stefannikolei) [inline tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs:480]:
```suggestion
```

**Pull request merged**